File: archive/r/r/longest-word.R
longest_word_length <- function(input_string) {
  # Checking input
  args <- commandArgs(trailingOnly = TRUE)
  
  if (length(args) == 0 || args == "") {
    cat("Usage: please provide a string\n")
  } else {
    # Split the string by whitespace and find the maximum word length in one line
    cat(max(nchar(unlist(strsplit(args[1], "[ \t\n\r]+")))), "\n")
  }
}

longest_word_length()

File: archive/r/ruby/LongestWord.rb
def longest_word_length(input)

    # Split the input by spaces
    words = input.split(" ")  

    # Start with 0 as the longest length
    max_length = 0        

    # Go through each word
    words.each do |word|       
      if word.length > max_length
        max_length = word.length
      end
    end
    
    # Return the length of the longest word
    max_length                 
  end

  # Check if the program has received input from the command line
  if ARGV.length > 0
    input_string = ARGV.join(" ")
    puts longest_word_length(input_string)
  else
    puts "Please provide an input string."
  end
  
  